llama.cpp

```diff
@@ -4276,6 +4336,21 @@ struct llm_build_context {
 
         ggml_build_forward_expand(gf, cur);
 
+#if defined(GGML_USE_KOMPUTE)
+        if (ctx_kompute) {
+            if (!ggml_vk_has_h2d_all(ctx_kompute)) {
+                ggml_vk_h2d_all(ctx_kompute);
+            } else {
+                ggml_vk_h2d_tensor(ctx_kompute, to_device_tensor);
+                ggml_vk_h2d_tensor(ctx_kompute, inp_pos);
+                ggml_vk_h2d_tensor(ctx_kompute, KQ_mask);
+                if (K_shift) {
+                    ggml_vk_h2d_tensor(ctx_kompute, K_shift);
+                }
+            }
+        }
+#endif
```

Comment:
I think we should try to move this logic outside the arch-specific graph building functions - the `ggml_vk_h2d_all()` should be in `llama_build_graph()` and the `ggml_vk_h2d_tensor()` should be in the `llm_build_cb` based on the tensor name.